crates/ruff_linter/src/message/text.rs

```diff
@@ -286,38 +283,61 @@ impl Display for MessageCodeFrame<'_> {
     }
 }
 
-fn replace_whitespace(source: &str, annotation_range: TextRange) -> SourceCode {
+/// Given some source code and an annotation range, this routine replaces
+/// tabs with ASCII whitespace, and unprintable characters with printable
+/// representations of them.
+///
+/// The source code returned has an annotation that is updated to reflect
+/// changes made to the source code (if any).
+fn replace_whitespace_and_unprintable(source: &str, annotation_range: TextRange) -> SourceCode {
     let mut result = String::new();
     let mut last_end = 0;
     let mut range = annotation_range;
     let mut line_width = LineWidthBuilder::new(IndentWidth::default());
 
+    let mut update_range = |index, len| {
+        if index < usize::from(annotation_range.start()) {
+            range += TextSize::new(len - 1);
+        } else if index < usize::from(annotation_range.end()) {
+            range = range.add_end(TextSize::new(len - 1));
+        }
+    };
+
+    // impl_show_nonprinting!(('\x07', "␇"), ('\x08', "␈"), ('\x1b', "␛"), ('\x7f', "␡"));
```

Comment:
is this actually commented out code that got missed, or just implying a theoretical macro to gesture to what this function is doing (if the latter, seems like too much detail)

Reply:
Nah just a snippet I copied from https://github.com/astral-sh/ruff/blob/beb8e2dfe07b71de5507fbff151e510fb284747a/crates/ruff_linter/src/text_helpers.rs

(That was what was originally being used to remove unprintable chars. Other code is still using it.)